app/services/submit_vehicle_service.rb

```diff
@@ -4,30 +4,20 @@ def self.call(cfe_estimate_id, cfe_session_data)
   end
 
   def call(cfe_estimate_id, cfe_session_data)
-    finance_form = Flow::Vehicle::FinanceHandler.model(cfe_session_data)
+    owned_model = Flow::Vehicle::OwnedHandler.model(cfe_session_data)
+    return unless owned_model.vehicle_owned
 
-    finance_form.vehicle_pcp.present? ? submit_vehicle_finance_data(cfe_estimate_id, cfe_session_data) : submit_vehicle_value_data(cfe_estimate_id, cfe_session_data)
-  end
-
-  def submit_vehicle_value_data(cfe_estimate_id, cfe_session_data)
-    value_form = Flow::Vehicle::ValueHandler.model(cfe_session_data)
-    return if value_form.vehicle_value.blank?
+    details_model = Flow::Vehicle::DetailsHandler.model(cfe_session_data)
 
-    cfe_connection.create_vehicle cfe_estimate_id, date_of_purchase: Time.zone.today.to_date,
-                                                   value: value_form.vehicle_value,
-                                                   loan_amount_outstanding: 0,
-                                                   in_regular_use: value_form.vehicle_in_regular_use
+    submit_vehicle_finance_data(cfe_estimate_id, details_model)
   end
 
-  def submit_vehicle_finance_data(cfe_estimate_id, cfe_session_data)
-    finance_form = Flow::Vehicle::FinanceHandler.model(cfe_session_data)
-    value_form = Flow::Vehicle::ValueHandler.model(cfe_session_data)
-    age_form = Flow::Vehicle::AgeHandler.model(cfe_session_data)
-    date_of_purchase = age_form.vehicle_over_3_years_ago ? 4.years.ago.to_date : 2.years.ago.to_date
+  def submit_vehicle_finance_data(cfe_estimate_id, model)
+    date_of_purchase = model.vehicle_over_3_years_ago ? 4.years.ago.to_date : 2.years.ago.to_date
     cfe_connection.create_vehicle cfe_estimate_id,
                                   date_of_purchase:,
-                                  value: value_form.vehicle_value,
-                                  loan_amount_outstanding: finance_form.vehicle_finance.presence,
-                                  in_regular_use: value_form.vehicle_in_regular_use
+                                  value: model.vehicle_value,
+                                  loan_amount_outstanding: model.vehicle_finance.presence || 0,
+                                  in_regular_use: model.vehicle_in_regular_use
```

Comment:
Think there is a tiny bug here - we should check vehicle_pcp being set, not the presence of vehicle_finance (in case they entered it and then changed their mind)

Reply:
Great spot! Thank you.